src/MICore/JsonLaunchOptions.cs

```diff
@@ -0,0 +1,392 @@
+        /// </summary>
+        [JsonProperty("program")]
+        public string Program { get; set; }
+
+        /// <summary>
+        /// The type of the engine. Must be "cppdbg".
+        /// </summary>
+        [JsonProperty("type", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string Type { get; set; }
+
+        /// <summary>
+        /// The architecture of the debuggee. This will automatically be detected unless this parameter is set. Allowed values are x86, arm, arm64, mips, x64, amd64, x86_64.
+        /// </summary>
+        [JsonProperty("targetArchitecture", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string TargetArchitecture { get; set; }
+
+        /// <summary>
+        /// .natvis file to be used when debugging this process. This option is not compatible with GDB pretty printing. Please also see "showDisplayString" if using this setting.
+        /// </summary>
+        [JsonProperty("visualizerFile", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string VisualizerFile { get; set; }
+
+        /// <summary>
+        /// When a visualizerFile is specified, showDisplayString will enable the display string. Turning this option on can cause slower performance during debugging.
+        /// </summary>
+        [JsonProperty("showDisplayString", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public bool? ShowDisplayString { get; set; }
+
+        /// <summary>
+        /// Indicates the console debugger that the MIDebugEngine will connect to. Allowed values are "gdb" "lldb".
+        /// </summary>
+        [JsonProperty("MIMode", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string MIMode { get; set; }
+
+        /// <summary>
+        /// The path to the mi debugger (such as gdb). When unspecified, it will search path first for the debugger.
+        /// </summary>
+        [JsonProperty("miDebuggerPath", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string MiDebuggerPath { get; set; }
+
+        /// <summary>
+        /// Network address of the MI Debugger Server to connect to (example: localhost:1234).
+        /// </summary>
+        [JsonProperty("miDebuggerServerAddress", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string MiDebuggerServerAddress { get; set; }
+
+        /// <summary>
+        /// Search stdout stream for server-started pattern and log stdout to debug output. Defaults to true.
+        /// </summary>
+        [JsonProperty("filterStdout", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public bool? FilterStdout { get; set; }
+
+        /// <summary>
+        /// Search stderr stream for server-started pattern and log stderr to debug output. Defaults to false.
+        /// </summary>
+        [JsonProperty("filterStderr", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public bool? FilterStderr { get; set; }
+
```

Comment:
@paulmaybee I moved it to LaunchOptions. I don't want to change the schema at this point since the current schema is what we have in the VSCode experience and I think our goal right now is to enable a similar scenario. 

When is the debugserver used? this isn't for the gdbserver connection as I think that one uses the miDebugServerAddress option.